core/proc/curry_spec.rb

```diff
@@ -159,10 +159,12 @@
   end
 
   it "can be passed more than _arity_ arguments if created from a proc" do
-    -> { @proc_add.curry(3)[1,2,3,4].should == 6 }.should_not
-      raise_error(ArgumentError)
-    -> { @proc_add.curry(1)[1,2].curry(3)[3,4,5,6].should == 6 }.should_not
-      raise_error(ArgumentError)
+    -> {
+      @proc_add.curry(3)[1,2,3,4].should == 6
+    }.should_not raise_error(ArgumentError)
+    -> {
+      @proc_add.curry(1)[1,2].curry(3)[3,4,5,6].should == 6
+    }.should_not raise_error(ArgumentError)
```

Comment:
Here it felt better to just write them in the style of most of these blocks, that resolves the rubocop issue too.

Reply:
It looks like this is an actual change in behaviour (I guess the old version parsed it as two statements), which would indicate this test is actually broken.

Reply:
I'm not sure why we even would want a `should_not raise_error` around a `should` statement, this feels superfluous here.

Reply:
Yeah `should_not raise_error` is an anti-pattern, let's remove these.